content/en/blog/2022/collector-builder-sample/index.md

```diff
@@ -0,0 +1,329 @@
+published by the OpenTelemetry community, these images can be as large as 40MB.
+This is due to all of the
+[receivers](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver),
+[processors](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor),
+and
+[exporters](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter)
+that are bundled into the image by default. With all of these default
+components, there is also the potential for security issues to arise, part of
+the reason why the
+[Collector maintainers recommend only enabling necessary components](https://github.com/open-telemetry/opentelemetry-collector/blob/5ab00fc/docs/security.md)
+in your Collector.
+
+The
+[collector-builder tool](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder)
+accelerates compiling stripped-down Collector images with a simple YAML config
+file. This file declares which components to include in the build, and
+collector-builder includes only those components (and nothing else). In a
+default Collector build, many more components are included in the compiled
+binary (even if they are not being enabled in the Collector’s runtime
+configuration). This is how you can achieve improvements over public images that
+include many excess components, with extra dependencies adding weight and
+security burden. Now, those extra components can’t take up space because they
+aren’t even available in the custom build.
+
+We’ve made this easier to use by including a sample config file in this
+repository which has been pre-filled with a few of the most relevant
+OpenTelemetry components for GCP. However, we encourage you to modify that
+sample file to fit your use case.
+
+For example, the following build file includes only the OTLP receiver and GCP
+exporter, along with a logging exporter:
+
+```
+receivers:
+  - import: go.opentelemetry.io/collector/receiver/otlpreceiver
+    gomod: go.opentelemetry.io/collector v0.57.2
+
+exporters:
+  - import: github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter
+    gomod: github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter v0.57.2
+  - import: go.opentelemetry.io/collector/exporter/loggingexporter
+    gomod: go.opentelemetry.io/collector v0.57.2
+```
+
+Edit
+[this file](https://github.com/GoogleCloudPlatform/opentelemetry-collector-builder-sample/blob/main/build/local/builder-config.yaml)
+in the repository and run `make build` to automatically generate a local binary,
+or `make docker-build` to compile a container image.
+
+Deploying a custom Collector cuts unneeded dependencies, making these
+trimmed-down builds are faster, lighter, and more secure. In this repository
+we’ve also included the steps required to set these builds up automatically with
+Cloud Build.
```

Comment:
Love this entire section, although this last paragraph feels a little too much like a promotion for cloud build. Is there a way to rework it to still mention that there's steps to publish, but perhaps caged with a "if you like..." or "if you are using..." wording? Otherwise I'd say it would be good to remove this.

Reply:
removed